### PR TITLE
Client stats

### PR DESCRIPTION
**Pull request by mkabischev**
Hi, this PR adds possibility to get information about number of active & idle connections.

### CODE REVIEW

**Comment by superstas** [comment]
Hello @bradfitz, could you please review this PR. 
The changes here would help us a lot.

Thank you!

File: memcache/memcache.go [hunk 1]
```diff
@@ -129,6 +129,12 @@ func NewFromSelector(ss ServerSelector) *Client {
 	return &Client{selector: ss}
 }
 
+// Stats contains statistic about connections being used by client.
```
Comment (bradfitz) [inline memcache/memcache.go]: statistics (plural)

File: memcache/memcache.go [hunk 2]
```diff
@@ -129,6 +129,12 @@ func NewFromSelector(ss ServerSelector) *Client {
...
+type Stats struct {
+	ActiveConns int
+	IdleConns   int
```
Comment (bradfitz) [inline memcache/memcache.go]: document these two also.

File: memcache/memcache.go [hunk 3]
```diff
@@ -144,9 +150,12 @@ type Client struct {
 	// be set to a number higher than your peak parallel requests.
 	MaxIdleConns int
 
+	// number of currently used connections
```
Comment (bradfitz) [inline memcache/memcache.go]: add docs that this must be accessed atomically

File: memcache/memcache.go [hunk 4]
```diff
@@ -465,6 +477,21 @@ func (c *Client) GetMulti(keys []string) (map[string]*Item, error) {
 	return m, err
 }
 
+// Stats returns current statistic
```
Comment (bradfitz) [inline memcache/memcache.go]: the current statistics. (with trailing period)

File: memcache/memcache.go [hunk 5]
```diff
@@ -144,9 +150,12 @@ type Client struct {
 	// be set to a number higher than your peak parallel requests.
 	MaxIdleConns int
 
+	// number of currently used connections
+	activeConns int32
+
 	selector ServerSelector
 
-	lk       sync.Mutex
+	lk       sync.RWMutex
```
Comment (bradfitz) [inline memcache/memcache.go]: doesn't seem worth it.

**Comment by dragonsinth** [comment]
Can I vote on this?  Or offer to address the feedback in this PR?  We also need this info for stats / monitoring.

File: memcache/memcache.go (hunk 5 above)
Comment (dragonsinth) [reply memcache/memcache.go:158]: maybe more straightforward to just make idleConns an atomic int as well and inc/dec when `freeconn` changes?

**Comment by dragonsinth** [comment]
@mkabischev are you opening a new one rebased instead of merged?

**Comment by dragonsinth** [comment]
Opened a version as https://github.com/bradfitz/gomemcache/pull/124

**Pull request closed without merging**